numpy/f2py/crackfortran.py

```diff
@@ -2873,6 +2906,84 @@ def analyzevars(block):
 analyzeargs_re_1 = re.compile(r'\A[a-z]+[\w$]*\Z', re.I)
 
 
+def param_eval(n, varsn, v, g_params, params):
+    # Creates a dictionary of indices and values for each parameter in a
+    # parameter array to be evaluated later.
+    #
+    # WARNING: It is not possible to initialize multidimensional array
+    # parameters e.g. dimension(-3:1, 4, 3:5) at this point. This is because in
+    # Fortran initialization through array constructor requires the RESHAPE
+    # intrinsic function. Since the right-hand side of the parameter declaration
+    # is not executed in f2py, but rather at the compiled c/fortran extension, later, it is not possible
+    # to execute a reshape of a parameter array.
+    # One issue remains: if the user wants to access the array parameter from python,
+    # we should either
+    # 1) allow them to access the parameter array using python standard indexing (which
+    #    is often incompatible with the original fortran indexing)
+    # 2) allow the parameter array to be accessed in python as a dictionary with fortran
+    #    indices as keys
+    # We are choosing 2 for now.
+    if any(a[:9] == 'dimension' for a in varsn['attrspec']):
+        # This is an array parameter.
+        # First, we parse the dimension information
+        a = [item[9:] for item in varsn['attrspec'] if item[:9] == 'dimension'][0]
```

Comment:
```suggestion
    dim_attrspecs = [item[9:] for item in varsn['attrspec'] if item[:9] == 'dimension']
    if len(dim_attrspecs) > 1:
        raise ValueError('dimension may appear at most once")
    elif dim_attrspecs:
        # This is an array parameter.
        # First, we parse the dimension information
        a = dim_attrspecs[0]
```

Reply:
The code above selects the first 9 characters of `varsn['attrspec']` to see if dimension is mentioned and then selects the dimension expression in `a` (something like a shape). Your suggestion selects the shape in `dim_attrspecs` and raises a `ValueError` if the shape has length > 1, which is not supposed to be a problem (for example if the declaration is `dimension(2)` then `dim_attrspecs == "(2)"` which has length 3. I guess what you were suggesting is that we don't have dimension specified twice in the declaration of this parameter?

Reply:
I don't believe your comment describes either what your original code does or what my suggestion does.

> guess what you were suggesting is that we don't have dimension specified twice in the declaration of this parameter?

Yes, this was my intent

Reply:
Sorry I had missed something. I see it now. Thanks for that!